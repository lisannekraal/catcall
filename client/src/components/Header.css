.navbar-brand {
  font-size: 1.5rem;
  font-weight: 300;
  display: flex;
  align-items: center;
  cursor: pointer;
}

.about-link {
  text-decoration: none;
  color: white;
}

.about-link button {
  height: 100%;
}

