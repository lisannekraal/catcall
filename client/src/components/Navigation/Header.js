import React, { useState, useEffect } from 'react';
import { HashLink as Link } from 'react-router-hash-link';
import { useHistory } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import { useQuery } from '@apollo/client';
import { GET_MODERATOR_BY_TOKEN } from '../../api/queries';

import SideDrawer from './SideDrawer';
import logo from '../../assets/Logo2.png'

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { List, Container, Hidden } from "@material-ui/core"
import { makeStyles } from '@material-ui/core/styles';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles((theme) => ({
  logo: {
    height: 42,
    width: 50,
    marginRight: 7,
  },
  navButton: {
    background: 'rgb(245, 37, 89)',
    borderRadius: 8,
    padding: '10px 15px',
    borderColor: '-internal-light-dark(rgb(118, 118, 118), rgb(133, 133, 133))',
  },
  navDisplayFlex: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  navDisplayLinks: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '50%',
    padding: '0 10px',
  }
}));

function Header({ removeCookie }) {

  let history = useHistory();
  const classes = useStyles();
  const [ navigations, setNavigations ] = useState([]);
  const { data } = useQuery(GET_MODERATOR_BY_TOKEN);

  let navLinksPermanent = [
    {
      title: 'Map',
      classN: 'fas fa-map-marked-alt',
      path: () => history.push({
        pathname: '/catcalls'
      })
    },
    {
      title: 'Community',
      classN: 'fab fa-instagram',
      path: () => window.open('https://www.instagram.com/catcallsofams/', '_blank')
    }
  ];

  useEffect(() => {
    if (data && data.getModeratorByToken) {
      setNavigations([...navLinksPermanent,
        {
          title: 'Dashboard',
          classN: 'fas fa-cog',
          path: () => history.push('/dashboard')
        },
        {
        title: 'Logout',
        classN: 'fas fa-user',
        path: () => {
          removeCookie('token');
          history.push('/');
        }
      }]);
    } else {
      setNavigations([...navLinksPermanent, {
          title: 'Moderator',
          classN: 'fas fa-cog',
          path: () => history.push('/login')
      }]);
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data]);

  return (
    <AppBar color='transparent' position="absolute" elevation={0} data-testid="navbar">
      <Toolbar style={{ color: 'white' }}>

        <Container className={classes.navDisplayFlex} maxWidth="xl">

          <div className="navbar-brand" onClick={() => history.push("/")}>
            <img src={logo} alt="logo" className={classes.logo} />
            <div className="catcall-font"> Catcalls of Amsterdam</div>
          </div>

          <Hidden mdDown>
            <List component="nav" className={classes.navDisplayLinks}>

              {/* 'About' nav component is seperate from loop as we use hash link package */}
              <Link
                to="/#about"
                className="about-link" >
                <Button key={uuidv4()} color='inherit' style={{textTransform: 'none', fontFamily: 'Arista Pro Alternate Bold', fontSize: '20px'}} startIcon={
                  <Icon className='fas fa-info-circle' fontSize="small" style={{ marginRight: 7 }} />
                } >About</Button>
              </Link>

              {navigations.map(({ title, classN, path }) => (
                <Button key={uuidv4()} style={{textTransform: 'none', fontFamily: 'Arista Pro Alternate Bold', fontSize: '20px'}} onClick={path} color='inherit' startIcon={
                  <Icon className={classN} fontSize="small" style={{ marginRight: 7 }} />
                }>{title}</Button>
              ))}
            </List>
          </Hidden>

          <Hidden mdDown>
            <Button style={{textTransform: 'none', fontSize: '16px'}} key={uuidv4()} color="inherit" className={classes.navButton} onClick={() => history.push("/catcalls/new")}>Report a new CatCall</Button>
          </Hidden>

        </Container>

        {/* Side drawer component for smaller screens */}
        <Hidden lgUp>
          <SideDrawer navLinks={navigations} />
        </Hidden>

      </Toolbar>
    </AppBar>
  )
}

export default Header;