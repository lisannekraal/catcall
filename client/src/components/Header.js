import React, { useState, useEffect } from 'react';
import { HashLink as Link } from 'react-router-hash-link';
import { useHistory } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';

import SideDrawer from './SideDrawer';
import './Header.css';
import logo from '../assets/Logo2.png'

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { List, Container, Hidden } from "@material-ui/core"
import { makeStyles } from '@material-ui/core/styles';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles((theme) => ({
  logo: {
    height: 42,
    width: 50,
    marginRight: 7,
  },
  navButton: {
    background: 'rgb(245, 37, 89)',
    borderRadius: 8,
    padding: '10px 15px',
    borderColor: '-internal-light-dark(rgb(118, 118, 118), rgb(133, 133, 133))',
  },
  navDisplayFlex: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  navDisplayLinks: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '50%',
    padding: '0 10px',
  }
}));

function Header({ token, removeCookie, setMod }) {

  let history = useHistory();
  const classes = useStyles();
  const [ navigations, setNavigations ] = useState([]);

  let navLinksPermanent = [
    {
      title: 'map',
      classN: 'fas fa-map-marked-alt',
      path: () => history.push({
        pathname: '/catcalls'
      })
    },
    {
      title: 'community',
      classN: 'fab fa-instagram',
      path: () => window.open('https://www.instagram.com/catcallsofams/', '_blank')
    }
  ];

  useEffect(() => {
    //depending on the mod token, show functionalities navbar

    if (token && token !== 'null') {
      setNavigations([...navLinksPermanent,
        {
          title: 'dashboard',
          classN: 'fas fa-cog',
          path: () => history.push('/dashboard')
        },
        {
        title: 'logout',
        classN: 'fas fa-user',
        path: () => {
          removeCookie('token');
          setMod(false);
          history.push('/');
        }
      }]);
    } else {
      setNavigations([...navLinksPermanent, {
          title: 'moderator',
          classN: 'fas fa-cog',
          path: () => history.push('/login')
      }]);
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [token]);

  return (
    <AppBar color='transparent' position="absolute" elevation={0} data-testid="navbar">
      <Toolbar style={{ color: 'white' }}>

        <Container className={classes.navDisplayFlex} maxWidth="xl">

          <div className="navbar-brand" onClick={() => history.push("/")}>
            <img src={logo} alt="logo" className={classes.logo} />
            <div className="catcall-font"> Catcalls of Amsterdam</div>
          </div>

          <Hidden mdDown>
            <List component="nav" className={classes.navDisplayLinks}>

              {/* 'About' nav component is seperate from loop as we use hash link package */}
              <Link
                to="/#about"
                className="about-link" >
                <Button key={uuidv4()} color='inherit' startIcon={<Icon className='fas fa-info-circle' fontSize="small" style={{ marginRight: 7 }} />} >About</Button>
              </Link>

              {navigations.map(({ title, classN, path }) => (
                <Button
                  key={uuidv4()}
                  onClick={path}
                  color='inherit'
                  startIcon={<Icon className={classN} fontSize="small" style={{ marginRight: 7 }} />}
                >{title}
                </Button>
              ))}
            </List>
          </Hidden>

          <Hidden mdDown>
            <Button  key={uuidv4()} color="inherit" className={classes.navButton} onClick={() => history.push("/catcalls/new")}>Report a new CatCall</Button>
          </Hidden>

        </Container>

        {/* Side drawer component for smaller screens */}
        <Hidden lgUp>
          <SideDrawer navLinks={navigations} />
        </Hidden>

      </Toolbar>
    </AppBar>
  )
}

export default Header;