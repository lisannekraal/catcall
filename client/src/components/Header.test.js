import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
import {render} from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

describe('Header tests:', ()=>{

  test('displays header correctly (large size screen)', ()=>{
    const theme = createMuiTheme({ props: {MuiWithWidth: {initialWidth: 'lg'}}});
    render( );

    const header = TestRenderer
      .create(<BrowserRouter><MuiThemeProvider theme={theme}><Header/></MuiThemeProvider></BrowserRouter>)
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  test('displays header correctly (small size screen)', ()=>{
    const theme = createMuiTheme({ props: {MuiWithWidth: {initialWidth: 'sm'}}});
    render( );

    const header = TestRenderer
      .create(<BrowserRouter><MuiThemeProvider theme={theme}><Header/></MuiThemeProvider></BrowserRouter>)
      .toJSON();
    expect(header).toMatchSnapshot();
  });

});