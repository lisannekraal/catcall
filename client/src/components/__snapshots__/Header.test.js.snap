// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests: displays header correctly (large size screen) 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorTransparent MuiPaper-elevation0"
  data-testid="navbar"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    style={
      Object {
        "color": "white",
      }
    }
  >
    <div
      className="MuiContainer-root makeStyles-navDisplayFlex-3 MuiContainer-maxWidthXl"
    >
      <div
        className="navbar-brand"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="makeStyles-logo-1"
          src="logowhite.png"
        />
        <div>
           Catcalls of Amsterdam
        </div>
      </div>
      <nav
        className="MuiList-root makeStyles-navDisplayLinks-4 MuiList-padding"
      >
        <a
          className="about-link"
          href="/#about"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root fas fa-info-circle MuiIcon-fontSizeSmall"
                  style={
                    Object {
                      "marginRight": 7,
                    }
                  }
                />
              </span>
              About
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fas fa-map-marked-alt MuiIcon-fontSizeSmall"
                style={
                  Object {
                    "marginRight": 7,
                  }
                }
              />
            </span>
            map
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fab fa-instagram MuiIcon-fontSizeSmall"
                style={
                  Object {
                    "marginRight": 7,
                  }
                }
              />
            </span>
            community
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fas fa-cog MuiIcon-fontSizeSmall"
                style={
                  Object {
                    "marginRight": 7,
                  }
                }
              />
            </span>
            moderator
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </nav>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navButton-2 MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Report a new CatCall
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</header>
`;

exports[`Header tests: displays header correctly (small size screen) 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorTransparent MuiPaper-elevation0"
  data-testid="navbar"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    style={
      Object {
        "color": "white",
      }
    }
  >
    <div
      className="MuiContainer-root makeStyles-navDisplayFlex-7 MuiContainer-maxWidthXl"
    >
      <div
        className="navbar-brand"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="makeStyles-logo-5"
          src="logowhite.png"
        />
        <div>
           Catcalls of Amsterdam
        </div>
      </div>
    </div>
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
